src/Gemini/Modules/Shell/Services/LayoutItemStatePersister.cs

```diff
@@ -121,30 +127,18 @@ public bool SaveState(IShell shell, IShellView shellView, string fileName)
             {
                 return false;
             }
-            finally
-            {
-                if (stream != null)
-                {
-                    stream.Dispose();
-                }
-            }
 
             return true;
         }
 
-        Type GetTypeFromContractNameAsILayoutItem(ExportAttribute attribute)
+        private static Type GetTypeFromContractNameAsILayoutItem(ExportAttribute attribute)
         {
-            if (attribute == null)
-                return null;
-
             string typeName;
             if ((typeName = attribute.ContractName) == null)
                 return null;
 
             var type = Type.GetType(typeName);
-            if (type == null || !typeof(ILayoutItem).IsInstanceOfType(type))
-                return null;
-            return type;
+            return typeof(ILayoutItem).IsAssignableFrom(type) ? type : null;
```

Comment:
I think it wasn't:
http://rextester.com/AQOBL12596

We implemented our custom LayoutItemStatePersister based on the original code, and spotted this bug.